alpha/declcfg/load.go

```diff
@@ -168,3 +168,18 @@ func readYAMLOrJSON(r io.Reader) (*DeclarativeConfig, error) {
 	}
 	return cfg, nil
 }
+
+// LoadFile will unmarshall declarative config components from a single 'file' located
+// at a filesystem hierarchy 'root' represented in string form as 'path'
+func LoadFile(root fs.FS, file io.Reader, path string) (*DeclarativeConfig, error) {
```

Comment:
Looking at this a bit closer, I'd suggest this function signature:
```suggestion
func LoadFile(root fs.FS, path string) (*DeclarativeConfig, error) {
```
where path is the path of the DC file we want to parse, relative to the root. We can do the `root.Open(path)` step inside `LoadFile`, I think. That would make it impossible for a caller is accidentally mess up sending in an io.Reader that isn''t loaded from the file at `path`.

Reply:
Yes, writing the godoc comment made this redundancy super clear.  I had planned to touch again in a later PR, but why not now? :)